src/core/include/openvino/core/type/float8_e8m0.hpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "openvino/core/core_visibility.hpp"
+
+namespace ov {
+
+/**
+ * @brief Class to represent the f8e8m0 type.
+ */
+class OPENVINO_API float8_e8m0 {
+public:
+    float8_e8m0() = default;
+    float8_e8m0(const float value);
+
+    template <typename I>
+    explicit float8_e8m0(I value) : float8_e8m0{static_cast<float>(value)} {}
+
+    template <typename T>
+    bool operator==(const T& other) const;
+    template <typename T>
+    bool operator!=(const T& other) const {
+        return !(*this == other);
+    }
+
+    template <typename T>
+    bool operator<(const T& other) const;
+    template <typename T>
+    bool operator<=(const T& other) const;
+    template <typename T>
+    bool operator>(const T& other) const;
+    template <typename T>
+    bool operator>=(const T& other) const;
+    template <typename T>
+    float8_e8m0 operator+(const T& other) const;
+    template <typename T>
+    float8_e8m0 operator+=(const T& other);
+    template <typename T>
+    float8_e8m0 operator-(const T& other) const;
+    template <typename T>
+    float8_e8m0 operator-=(const T& other);
+    template <typename T>
+    float8_e8m0 operator*(const T& other) const;
+    template <typename T>
+    float8_e8m0 operator*=(const T& other);
+    template <typename T>
+    float8_e8m0 operator/(const T& other) const;
+    template <typename T>
+    float8_e8m0 operator/=(const T& other);
+
+    operator float() const;
+
+    static constexpr float8_e8m0 from_bits(uint8_t bits) {
+        return float8_e8m0(bits, true);
```

Comment:
```suggestion
        return float8_e8m0(bits ? bits : 1, true);
```
Consider if check for bits for illegal value has to be done or  convert 0 -> 1.